auth/api/iam/user.go

```diff
@@ -102,35 +71,35 @@ func (r Wrapper) handleUserLanding(echoCtx echo.Context) error {
 		return echoCtx.NoContent(http.StatusForbidden)
 	}
 	accessTokenRequest := redirectSession.AccessTokenRequest
+
+	verifier, err := did.ParseDID(accessTokenRequest.Body.Verifier)
+	if err != nil {
+		return err
+	}
+
 	// burn token
 	err = store.Delete(token)
 	if err != nil {
 		//rare, log just in case
 		log.Logger().WithError(err).Warn("delete token failed")
 	}
-	// create UserSession with userID from request
+	// create oauthSession with userID from request
 	// generate new sessionID and clientState with crypto.GenerateNonce()
-	userSession := UserSession{
+	oauthSession := OAuthSession{
 		ClientState: crypto.GenerateNonce(),
-		SessionID:   crypto.GenerateNonce(),
-		UserID:      *accessTokenRequest.Body.UserID, // should be there...
-		OwnDID:      redirectSession.OwnDID,
+		OwnDID:      &redirectSession.OwnDID,
+		VerifierDID: verifier,
+		FlowToken:   redirectSession.FlowToken,
```

Comment:
flow/session seem to be 2 names for the same thing?

Reply:
We can settle on session I think,